src/Microsoft.AspNetCore.JsonPatch/Converters/JsonPatchDocumentConverter.cs

```diff
@@ -57,7 +57,7 @@ public override bool CanConvert(Type objectType)
             }
             catch (Exception ex)
             {
-                throw new JsonPatchException(Resources.FormatInvalidJsonPatchDocument(objectType.Name), ex);
+                throw new JsonSerializationException(Resources.FormatInvalidJsonPatchDocument(), ex);
```

Comment:
nit: you can use `Resources.InvalidJsonPatchDocument` since there are no parameters.